File: LeetCode/1021. Remove Outermost Parentheses/solution.cpp
/**
 * 59 / 59 test cases passed.
 * Runtime: 0 ms
 * Memory Usage: 6.5 MB 
 */
class Solution {
public:
    string removeOuterParentheses(string s) {
        string ans = "";
        stack<char> stk;
        for (auto &c : s) {
            if (c == ')') stk.pop();
            if (stk.size()) ans.push_back(c);
            if (c == '(') stk.push(c);
        }
        return ans;
    }
};

File: LeetCode/1021. Remove Outermost Parentheses/solution.py
"""
59 / 59 test cases passed.
Runtime: 44 ms
Memory Usage: 15.2 MB
"""
class Solution:
    def removeOuterParentheses(self, s: str) -> str:
        count = 0
        ans = ''
        for c in s:
            if c == ')':
                count -= 1
            if count != 0:
                ans += c
            if c == '(':
                count += 1
        return ans
